### PR TITLE
feat(kselect): introduce kui tokens [KHCP-7722]

### PR DESCRIPTION
**Pull request by portikM**
# Summary

Addresses: https://konghq.atlassian.net/browse/KHCP-7722

Changes:
- introduces `kui-` design tokens in `KSelect` component
- removes any usage of Kongponents utility classes in `KSelect` component template

## PR Checklist

* [ ] Does not introduce dependencies
* [ ] **Functional:** all changes do not break existing APIs and if so, bump major version.
* [ ] **Tests pass:** check the output of yarn test
* [ ] **Naming:** the files and the method and prop variables use the same naming conventions as other Kongponents
* [ ] **Framework style:** abides by the essential rules in Vue's style guide
* [ ] **Cleanliness:** does not have formatting issues, unused code (e.g., console.logs, debugger), or leftover comments
* [ ] **Docs:** includes a technically accurate README, uses JSDOC where appropriate

### CODE REVIEW

File: src/components/KSelect/KSelect.vue [hunk 1]
```diff
@@ -40,14 +40,14 @@
...
-          class="clear-selection-icon cursor-pointer non-visual-button"
+          class="clear-selection-icon"
           @click="clearSelection"
           @keyup.enter="clearSelection"
         >
           <KIcon
             color="var(--blue-200)"
             icon="clear"
-          />
+          /><!-- TODO: token needed -->
```
Comment (portikM) [inline src/components/KSelect/KSelect.vue:50]: @adamdehaven any idea what can we do here? Appropriate `color-text-primary-...` token variant doesn't exist and `$tmp-` SCSS won't work in template (unless used with interpolation?)

Comment (adamdehaven) [reply src/components/KSelect/KSelect.vue:50]: You're saying there's not a token for `blue-200`, correct?

If that's the case, I would pick the _closest existing new `kui` token_ and use that, with its fallback, instead. 

Changing the icon color a little isn't going to break anything.

Comment (portikM) [reply src/components/KSelect/KSelect.vue:50]: There is no close value color text token (closest one is `color-text-primary` which is noticeably different). There are close tokens for background color, but we decided against using background color tokens for icons color. I would add a temp SCSS token for this so we can decide later, but like I said it won't work.

Comment (adamdehaven) [reply src/components/KSelect/KSelect.vue:50]: Just use the closest and put a screenshot in the PR

Comment (portikM) [reply src/components/KSelect/KSelect.vue:50]: Here are before and after screenshots
Screen Shot 2023-07-18 at 11 13 25 AM
Screen Shot 2023-07-18 at 11 16 17 AM

Comment (adamdehaven) [reply src/components/KSelect/KSelect.vue]: I think the easiest thing for this particular scenario is to hard-code the icon color to the hex value and just drop CSS variable support. I don’t believe we’re customizing he select icon color here anyway

File: src/components/KSelect/KSelect.vue [hunk 2]
```diff
@@ -1059,14 +1070,19 @@ $chevronDownIconMargin: 10px;
...
-      top: 0;
+      top: var(--kui-space-0, $kui-space-0);
     }
 
     .k-popover-content {
       @include kSelectPopoverMaxHeight;
 
+      .k-select-list {
+        margin: var(--kui-space-0, kui-space-0) !important;
+        padding: var(--kui-space-0, kui-space-0) !important;
```
Comment (adamdehaven) [inline src/components/KSelect/KSelect.vue:1083]: issue: missing `$` here

File: src/components/KSelect/KSelectItem.vue [hunk 3]
```diff
@@ -99,60 +104,62 @@ const handleClick = (e: MouseEvent): void => {
...
 
     .k-select-item-label {
-      color: var(--grey-600);
-      font-size: 14px;
-      font-weight: 500;
-      line-height: 20px;
-      padding: 8px;
+      color: var(--grey-600, var(--kui-color-text-primary-strong, $kui-color-text-primary-strong));
+      font-size: var(--kui-font-size-30, $kui-font-size-30);
+      font-weight: var(--kui-font-weight-regular, $kui-font-weight-regular);
```
Comment (adamdehaven) [inline src/components/KSelect/KSelectItem.vue:109]: issue: regular font weight is `400` -- you need `500` here

**Review by adamdehaven — CHANGES_REQUESTED** [review]

File: src/components/KSelect/KSelectItem.vue (hunk 3 above)
Comment (portikM) [reply src/components/KSelect/KSelectItem.vue]: Fixed

File: src/components/KSelect/KSelect.vue (hunk 2 above)
Comment (portikM) [reply src/components/KSelect/KSelect.vue]: Fixed

**Review by adamdehaven — APPROVED** [review]

**Review by adamdehaven — APPROVED** [review]

**Pull request merged**